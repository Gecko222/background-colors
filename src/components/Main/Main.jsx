import React, { Component } from 'react';

/**
 * main component
 */
class Main extends Component {
	/**
     * render
     * @return {ReactElement}
     */
	render() {
		return <h1>Hello, world!</h1>;
	}
}

export default Main;
