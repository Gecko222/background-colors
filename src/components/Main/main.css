.content {
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	display: flex;
	flex-direction: row;
	height: 30px;
}
