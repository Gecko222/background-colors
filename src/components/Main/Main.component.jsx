import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { map } from 'lodash';

import './main.css';

import AcceptButton from '../accept-button/accept-button.component';
import Autocomplete from '../autocomplete/autocomplete.component';
import { fetchColors } from '../../actions/colors.action';

/**
 * main component
 */
class Main extends Component {
	/**
	 * render
	 * @return {ReactElement}
	 */
	render() {
		return (
			<div className="content">
				{
					this.props.fetched ?
						this.props.fetchError ? this._renderError() :
							this._renderContent() :
						this._renderWait()
				}
			</div>
		);
	}

	/**
	 * component did mount.
	 * fetch colors from source
	 */
	componentDidMount() {
		this.props.fetchColors();
	}

	/**
	 * render wait information
	 * @return {ReactElement}
	 */
	_renderWait() {
		return <div>Loading data, please wait...</div>;
	}

	/**
	 * render error information
	 * @return {ReactElement}
	 */
	_renderError() {
		return <div>Failed to get data, please try again later.</div>;
	}

	/**
	 * render app content
	 * @return {ReactFragment}
	 */
	_renderContent() {
		return <Fragment>
			<Autocomplete items={this._getItems()} />
			<AcceptButton />
		</Fragment>;
	}

	/**
	 * get items list
	 * @return {Object[]}
	 */
	_getItems() {
		return map(this.props.colors, color => color.name);
	}

	static propTypes = {
		fetchColors: PropTypes.func.isRequired,
		fetched: PropTypes.bool.isRequired,
		fetchError: PropTypes.bool.isRequired,
		colors: PropTypes.arrayOf(PropTypes.shape({
			name: PropTypes.string,
			hes: PropTypes.string,
		})),
	}
}

const mapStateToProps = ({ appState }, props) => {
	return {
		fetched: appState.fetched,
		fetchError: appState.fetchError,
		colors: appState.colors || [],
	};
};

const mapDispatchToProps = dispatch => {
	return {
		fetchColors: () => fetchColors(dispatch),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Main);
