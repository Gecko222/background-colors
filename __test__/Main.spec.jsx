import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

global.fetch = require('jest-fetch-mock');

import Main from '../src/components/main/main.component';

// Snapshot for Home React Component
describe('>>>Main --- Snapshot', () => {
	const initialState = {
		appState: {
			fetched: false,
			fetchError: false,
			colors: [],
		},
		autocomplete: {},
	};
    const mockStore = configureStore();

	it('+++capturing Snapshot of Main loading', () => {
		const store = mockStore(initialState);
		const wrapper = mount( <Provider store={store}><Main /></Provider> );

		const renderedValue = renderer.create(wrapper).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});

	it('+++capturing Snapshot of Main with error', () => {
		const store = mockStore({
			...initialState,
			appState: {
				...initialState.appState,
				fetched: true,
				fetchError: true,
			},
		});
		const wrapper = mount( <Provider store={store}><Main /></Provider> );

		const renderedValue = renderer.create(wrapper).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});

	it('+++capturing Snapshot of Main', () => {
		const store = mockStore({
			...initialState,
			appState: {
				...initialState.appState,
				fetched: true,
				fetchError: false,
			},
		});
		const wrapper = mount( <Provider store={store}><Main /></Provider> );

		const renderedValue = renderer.create(wrapper).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
