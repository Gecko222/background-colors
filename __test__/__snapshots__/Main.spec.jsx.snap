// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Main --- Snapshot +++capturing Snapshot of Main 1`] = `
<div
  className="content"
>
  <div
    className="auto-complete-wrapper"
  >
    <input
      className="auto-complete-field"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </div>
  <button
    className="accept-button"
    onClick={[Function]}
  >
    Accept
  </button>
</div>
`;

exports[`>>>Main --- Snapshot +++capturing Snapshot of Main loading 1`] = `
<div
  className="content"
>
  <div>
    Loading data, please wait...
  </div>
</div>
`;

exports[`>>>Main --- Snapshot +++capturing Snapshot of Main with error 1`] = `
<div
  className="content"
>
  <div>
    Failed to get data, please try again later.
  </div>
</div>
`;
